lib/uptime/env.ex

```diff
@@ -27,23 +29,82 @@ defmodule Uptime.Env do
     :ok
   end
 
+  @doc """
+  Read a value from the environment.
+  First check for a secrets file, then the environment, then the config file.
+  secrets file > environment variable > config file
+  """
   @spec read(String.t()) :: String.t()
   def read(name) do
-    case File.read("/etc/secrets/#{name}") do
-      {:ok, value} -> value
-      _not_found -> System.get_env(name)
+    prefixed_name = "#{@prefix}_#{name}"
+
+    file_contents =
+      :supported_secrets_dir
+      |> get("/etc/secrets")
+      |> Path.join(prefixed_name)
+      |> File.read()
```

Comment:
just a thought but it took me a minute to process what this was doing, it might be worth breaking out to a variable or module attribute

```suggestion
    prefixed_name = "#{@prefix}_#{name}"
    
    # this could also be a module attr since this is compile time
    # @secrets_directory Application.compile_env(:uptime, :supported_secret_dir)
    secrets_directory = get(:supported_secrets_dir, "/etc/secrets")

    secret_file_contents =
      Path.join(secrets_directory, prefixed_name)
      |> File.read()
```

Reply:
Ya I can do that